src/EnergyPlus/OutputReportTabular.cc

```diff
@@ -10785,6 +10802,166 @@ namespace OutputReportTabular {
 		}
 	}
 
+    // Parses the contents of the EIO (initializations) file and creates subtables for each type of record in the tabular output files
+	// Glazer - November 2016
+	void
+	WriteEioTables() {
+
+		if (displayEioSummary)
+		{
+			// all arrays are in the format: (row, column)
+			Array1D_string columnHead;
+			Array1D_int columnWidth;
+			Array1D_string rowHead;
+			Array2D_string tableBody;
+			Array1D_int colUnitConv;
+
+			// setting up  report header 
+			WriteReportHeaders("Initialization Summary", "Entire Facility", isAverage);
+
+			// since the EIO initilization file is open at this point must close it to read it and then reopen afterward.
+			gio::close(OutputFileInits); 		// for testing only
+
+			std::ifstream eioFile;
```

Comment:
@Myoldmopar I actually spent a few days trying to figure out an approach that would work more like what you suggest.  There are 822 lines that write to the EIO file so I wanted to make a change that tried to keep the write lines as similar as possible to how they are now but instead of writing directly would call a function/method. Basically the same approach as you are suggesting. I ran into a few problems. The heavy use of Fortran style Formats for almost every line means that I would need to overload the gio::write and << operator. I tried to work on this but did not get far. I also tried another approach using stringstream figuring I could just leverage the Fortran style Format handling from the objexx library but did not get vary far. I started planning some meta code with Python that would get rid of all the Fortran Formats and replace them with normal C++ output manipulators. I also a little exploration of how C++ functions could accept any number of arguments of any types and that seemed impossible. It is a tough nut to crack.